Index: base/plugins.cpp
===================================================================
--- base/plugins.cpp	(revision 45936)
+++ base/plugins.cpp	(working copy)
@@ -118,6 +118,9 @@
 		#if PLUGIN_ENABLED_STATIC(KYRA)
 		LINK_PLUGIN(KYRA)
 		#endif
+		#if PLUGIN_ENABLED_STATIC(LASTEXPRESS)
+		LINK_PLUGIN(LASTEXPRESS)
+		#endif
 		#if PLUGIN_ENABLED_STATIC(LURE)
 		LINK_PLUGIN(LURE)
 		#endif
Index: configure
===================================================================
--- configure	(revision 45936)
+++ configure	(working copy)
@@ -90,6 +90,7 @@
 add_engine groovie2 "Groovie 2 games" no
 add_engine kyra "Legend of Kyrandia" yes "lol"
 add_engine lol "Lands of Lore" no
+add_engine lastexpress "The Last Express" no
 add_engine lure "Lure of the Temptress" yes
 add_engine m4 "M4/MADS" no
 add_engine made "MADE" yes
Index: engines/engines.mk
===================================================================
--- engines/engines.mk	(revision 45936)
+++ engines/engines.mk	(working copy)
@@ -69,6 +69,11 @@
 endif
 endif
 
+ifdef ENABLE_LASTEXPRESS
+DEFINES += -DENABLE_LASTEXPRESS=$(ENABLE_LASTEXPRESS)
+MODULES += engines/lastexpress
+endif
+
 ifdef ENABLE_LURE
 DEFINES += -DENABLE_LURE=$(ENABLE_LURE)
 MODULES += engines/lure
